source/module_base/lapack_connector.h

```diff
@@ -252,6 +291,38 @@ class LapackConnector
         delete[] bux;
         delete[] zux;
 
+    }
+    // wrap function of fortran lapack routine zheev.
+    static inline
+    void zhegvx( const int itype, const char jobz, const char range, const char uplo,
+                 const int n, const std::complex<double>* a, const int lda, const std::complex<double>* b,
+                 const int ldb, const double vl, const double vu, const int il, const int iu,
+                 const double abstol, const int m, double* w, std::complex<double>* z, const int ldz,
+                 std::complex<double>* work, const int lwork, double* rwork, int* iwork,
+                 int* ifail, int info, int nbase_x)
+    {
+        // Transpose the std::complex matrix to the fortran-form real-std::complex array.
+        std::complex<double>* aux = LapackConnector::transpose(a, n, lda, nbase_x);
+        std::complex<double>* bux = LapackConnector::transpose(b, n, ldb, nbase_x);
+        std::complex<double>* zux = new std::complex<double>[n*iu];// mohan modify 2009-08-02
+
+        // call the fortran routine
+        zhegvx_(&itype, &jobz, &range, &uplo, &n, aux, &lda, bux, &ldb, &vl,
+                &vu, &il,&iu, &abstol, &m, w, zux, &ldz, work, &lwork, rwork, iwork, ifail, &info);
+
+        // Transpose the fortran-form real-std::complex array to the std::complex matrix
+        for (int i = 0; i < iu; ++i)
+        {
+            for (int j = 0; j < n; ++j)
+            {
+                z[j * nbase_x + i] = zux[i*n+j];
+            }
+        }
+
+        // free the memory.
+        delete[] aux;
+        delete[] bux;
+        delete[] zux;
```

Comment:
I just wonder what does 'bux' means?

Reply:
This variable is named to maintain consistency with [the previous code](https://github.com/deepmodeling/abacus-develop/blob/adfbcb25171509dde22b69331175d20b203cc61e/source/module_base/lapack_connector.h#L252). So I don't know what he means 😂.

Reply:
Well, I guess aux for auxiliary😂
Conforming to tradition is not always good in ABACUS

Reply:
If the data in the entire CG process is changed to column-first in the near future, this function will be deprecated. So I did not optimize it here.

Reply:
> If the data in the entire CG process is changed to column-first in the near future, this function will be deprecated. So I did not optimize it here.

That's OK! It would be better submitting an issue on this extra transpose required for row-major storage.